pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_reconcile.go

```diff
@@ -159,18 +161,33 @@ func (c *controller) reconcilePermissionClaims(ctx context.Context, apiBinding *
 		}
 	}
 
-	klog.V(2).Infof("Adding permission claims %v for APIBinding %s", addedPermissionClaims, fmt.Sprintf("%s|%s", lc, apiBinding.Name))
+	if len(addedPermissionClaims) != 0 {
+		klog.V(2).Infof("Adding permission claims %v for APIBinding %s", addedPermissionClaims, fmt.Sprintf("%s|%s", lc, apiBinding.Name))
+	}
 
 	var allPatchErrors []error
 	for gvr, claims := range addedPermissionClaims {
 		for _, claim := range claims {
 			//TODO: use indexer instead of lister.
-			objs, err := claim.lister.Lister().List(labels.Everything())
+			if !claim.informer.Informer().HasSynced() {
+				return fmt.Errorf("unable sync cache")
+			}
+			// objs, err := claim.informer.Informer().GetIndexer().ByIndex(indexers.ByLogicalCluster, lc.String())
+			// if err != nil {
+			// 	return err
+			// }
+
+			objs, err := c.dynamicClusterClient.Resource(gvr).List(logicalcluster.WithCluster(ctx, lc), metav1.ListOptions{})
 			if err != nil {
 				return err
 			}
 			var errs []error
-			for _, obj := range objs {
+			klog.V(4).Infof("udpating %v objs of type: %v.%v", len(objs.Items), gvr.Resource, gvr.Group)
+			for _, obj := range objs.Items {
+				// o, ok := obj.(runtime.Object)
+				// if !ok {
+				// 	return fmt.Errorf("expected type %v got %T", "runtime.Object", obj)
+				// }
 				newObj := obj.DeepCopyObject()
```

Comment:
this is a super heavy operation. As you are sending an empty patch now, there is no need.

Reply:
updated